byway/src/main/webapp/track-interests.js

```diff
@@ -26,23 +26,31 @@ if(document.readyState === 'loading') {
  * and the form tracking buttons selected.
  */
 function loadContent() {
-  loadId();
-  loadButtons();
-  loadForm();
+  getTripIdFromUrl();
+  loadButtonsWithInterests();
+  loadFormToSaveInterests();
 }
 
-/* Parse url to retrieve the trip id. */
-function loadId() {
+/* Parse url to retrieve the trip id and load the next page with it. */
+function getTripIdFromUrl() {
   let params = new URLSearchParams(location.search);
   tripId = params.get('tripId');
+  if(tripId === null) {
+    // Send back to page where a trip can be made
+    alert("tripId not created! Create a new trip.");
+    window.location.href = '/index.html';
+  } else {
+    let nextPage = document.getElementById("next-page");
+    nextPage.href = "/generator.html?" + new URLSearchParams({tripId}).toString();
+  }
 }
 
 /**
  * Sets properties to the interest form on the html
  * page. Handles submit event to keep user on the
- * same page and send information.
+ * same page and send information through a fetch request.
  */
-function loadForm() {
+function loadFormToSaveInterests() {
```

Comment:
`load` implies the function itself loads data from the server, but this just configures what is essentially the "save" button.

Reply:
Removed the form